src/terminal/terminal.c

```diff
@@ -777,31 +778,43 @@ void guac_terminal_commit_cursor(guac_terminal* term) {
 
     guac_terminal_char* guac_char;
 
-    guac_terminal_buffer_row* old_row;
-    guac_terminal_buffer_row* new_row;
+    guac_terminal_buffer_row* row;
 
     /* If no change, done */
     if (term->visible_cursor_row == term->cursor_row && term->visible_cursor_col == term->cursor_col)
```

Comment:
Will this potentially result in changes to cursor visibility not taking effect until cursor is moved?

Reply:
I think you are right, we would also need to check `term->cursor_visible == (term->visible_cursor_col == -1)` or `term->cursor_visible == term->visible_cursor_visible` (see the other comment).